orttraining/orttraining/training_api/include/module.h

```diff
@@ -91,13 +91,23 @@ struct Module {
   // Returns the output count for eval graph
   size_t GetEvalModeOutputCount() const noexcept;
 
+  // Return size of all parameters
+  size_t GetParametersSize(const bool trainable_only=true) const noexcept;
+
+  // Copy parameters onto contiguous buffer held by parameters_buffer
+  Status CopyParametersToBuffer(OrtValue& parameters_buffer, const bool trainable_only=true);
+  
+  // Copy parameter values from contiguous buffer held by parameters_buffer onto parameters
+  Status CopyBufferToParameters(OrtValue& parameters_buffer, const bool trainable_only=true);
+
```

Comment:
Since we need to maintain the order in which the parameters are extracted/fetched into, we need to define it over either the module or optimizer. Module makes more logical sense.